torch/distributed/fsdp/_flat_param.py

```diff
@@ -2707,15 +2707,15 @@ def _warn_use_fake_reduce(log: logging.Logger, warning: str):
 
 @torch._dynamo.allow_in_graph
 def _same_storage(a, b):
-    return a._typed_storage()._data_ptr() == b._typed_storage().data_ptr()
+    return a.untyped_storage()._data_ptr() == b.untyped_storage()._data_ptr()
```

Comment:
@awgu not sure what the type of the second value is, but why before was it "_data_ptr()" for a and "data_ptr()" for b?

Reply:
It looks like it was valid since both are defined with the public one calling into the private one:
![Screenshot 2024-01-03 at 6 35 16 PM](https://github.com/pytorch/pytorch/assets/31054793/ae0c5c53-70f8-4545-a948-d54d52e21a4b)